bytestring_splitter/__init__.py

```diff
@@ -44,18 +48,33 @@ def set_original_bytes_repr(self, bytes_representation):
         self._original_bytes = bytes_representation
 
     def finish(self):
-        finished_values = {}
+
         for message_name, (bytes_for_message, message_class, kwargs) in self.processed_objects.items():
-            finished_values[message_name] = produce_value(message_class,
+            self._finished_values[message_name] = produce_value(message_class,
                                                           message_name,
                                                           bytes_for_message,
                                                           kwargs)
-        return self._receiver(**finished_values, **self._additional_kwargs)
+        return self._receiver(**self._finished_values, **self._additional_kwargs)
+
+    def __getattr__(self, message_name):
+        # First we'll try to see if this message_name already has a finished value:
+        try:
+            return self._finished_values[message_name]
+        except KeyError:  # suppress might be good here, but it appears to have a performance penalty, and this is a performance-concerned function.
+            pass
+
+        try:
+            bytes_for_message, message_class, kwargs = self.processed_objects[message_name]
+            self._finished_values[message_name] = produce_value(message_class,
+                                                               message_name,
+                                                               bytes_for_message,
+                                                               kwargs)
```

Comment:
An extremely minor comment: My obsessive-compulsive self tells me that both calls to `produce_value()` (here and L53) have the arguments not well aligned.